### PR TITLE
Co 1661 tests

### PR DESCRIPTION
**Pull request by zextrasGiovanniDefacci**

### CODE REVIEW

**Review by aheeva-yuliya — APPROVED** [review]

**Review by sonersivri — COMMENTED** [review]
@zextrasGiovanniDefacci I have some comments

File: store/src/test/java/com/zimbra/cs/account/ProvUtilTest.java [hunk 1]
```diff
@@ -503,4 +554,256 @@ void searchGal_fails_when_using_ldap_backend_and_unsupported_arguments_passed2()
     Assertions.assertEquals(expected, stdErr.toString());
   }
 
+  @ParameterizedTest
+  @ValueSource(strings = {
+          "addAccountLogger",
+          "addAccountLogger test@test.com status",
+          "addAccountLogger test@test.com zimbra.lmtp warn anotherParam par4 par5 par6",
+  })
+  void addAccountLoggerFailsOnInvalidParameterCount(String cmd) throws Exception {
```
[the remark this answers is not in the document]
Comment (zextrasGiovanniDefacci) [reply store/src/test/java/com/zimbra/cs/account/ProvUtilTest.java:568]: It's not that easy, the program ends with system exit, so it is quite difficult to do something after an system exit happens

File: store/src/main/java/com/zimbra/cs/account/Console.java [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+      writer.write(text + "\n");
+      writer.flush();
+    } catch (IOException e) {
+      ps.println(text);
+    }
+  }
+
+  public void printOutput(String text) {
+    try {
+      BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(this.stdOut, Charsets.UTF_8));
```
[the remark this answers is not in the document]
Comment (zextrasGiovanniDefacci) [reply store/src/main/java/com/zimbra/cs/account/Console.java:55]: We did it in the [refactor branch](https://github.com/zextras/carbonio-mailbox/tree/CO-1661-refactor): [store/src/main/java/com/zimbra/cs/account/Console.java](https://github.com/zextras/carbonio-mailbox/blob/85ab12f7314cfe52fc4c347f803f2a27d8c849f7/store/src/main/java/com/zimbra/cs/account/Console.java)

File: store/src/main/java/com/zimbra/cs/account/Console.java [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+  }
+
+  public void printStacktrace(Exception exception) {
+    exception.printStackTrace(this.stdError);
+  }
+
+  public void printError(String text) {
+    PrintStream ps = this.stdError;
+    try {
+      BufferedWriter writer = new BufferedWriter(new OutputStreamWriter(ps, Charsets.UTF_8));
```
[the remark this answers is not in the document]
Comment (zextrasGiovanniDefacci) [reply store/src/main/java/com/zimbra/cs/account/Console.java:45]: We did it in the [refactor branch](https://github.com/zextras/carbonio-mailbox/tree/CO-1661-refactor): [store/src/main/java/com/zimbra/cs/account/Console.java](https://github.com/zextras/carbonio-mailbox/blob/85ab12f7314cfe52fc4c347f803f2a27d8c849f7/store/src/main/java/com/zimbra/cs/account/Console.java)

File: store/src/test/java/com/zimbra/cs/account/ProvUtilTest.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (aheeva-yuliya) [reply store/src/test/java/com/zimbra/cs/account/ProvUtilTest.java:568]: I added the assertion

**Review by sonersivri — APPROVED** [review]

**Pull request merged**